docs/writing-shaders.md

```diff
@@ -39,23 +39,42 @@ void main(void) {
 
 ## Uniforms
 
-### `layerIndex` uniform
+### Viewport uniforms
+
+#### `mat4 modelMatrix`
+
+The user supplied model matrix if `props.modelMatrix` is set on the layer,
+otherwise an identity matrix.
+
+#### `mat4 viewMatrix`
+
+The view matrix of the current viewport.
+
+#### `vec2 viewportSize`
+
+Viewport width and height. Useful when rendering pixel sizes.
```

Comment:
We should be really crisp about what coordinates this is specified in. Device coordinates or Window coordinates. I assume the former.